tests/meta/demultiplex/test_utils.py

```diff
@@ -1,22 +1,137 @@
 from pathlib import Path
 
 import pytest
-
 from cg.constants.constants import FileExtensions
 from cg.constants.demultiplexing import DemultiplexingDirsAndFiles
 from cg.constants.sequencing import FLOWCELL_Q30_THRESHOLD, Sequencers
 from cg.exc import FlowCellError
 from cg.meta.demultiplex.utils import (
+    add_flow_cell_name_to_fastq_file_path,
     create_delivery_file_in_flow_cell_directory,
     get_lane_from_sample_fastq,
     get_q30_threshold,
     get_sample_sheet_path,
+    is_file_path_compressed_fastq,
+    is_lane_in_fastq_file_name,
+    is_sample_id_in_directory_name,
+    is_valid_sample_fastq_file,
     parse_flow_cell_directory_data,
-    add_flow_cell_name_to_fastq_file_path,
 )
 from cg.models.demultiplex.flow_cell import FlowCellDirectoryData
 
 
+def test_validate_sample_fastq_with_valid_file():
+    # GIVEN a sample fastq file with a lane number and sample id in the parent directory name
+    sample_fastq = Path("Sample_123/sample_L0002.fastq.gz")
+
+    # WHEN validating the sample fastq file
+    is_valid = is_valid_sample_fastq_file(sample_fastq=sample_fastq, sample_internal_id="sample")
+
+    # THEN it should be valid
+    assert is_valid
+
+
+def test_validate_sample_fastq_without_sample_id_in_parent_directory_name():
+    # GIVEN a sample fastq file without a sample id in the parent directory name or file name
+    sample_fastq = Path("L0002.fastq.gz")
+
+    # WHEN validating the sample fastq file
+    is_valid_fastq = is_valid_sample_fastq_file(
+        sample_fastq=sample_fastq, sample_internal_id="sample_id"
+    )
+
+    # THEN it should not be valid
+    assert not is_valid_fastq
+
+
+def test_validate_sample_fastq_without_lane_number_in_path():
+    # GIVEN a sample fastq file without a lane number
+    sample_fastq = Path("Sample_123/sample_id.fastq.gz")
+
+    # WHEN validating the sample fastq file
+    is_valid_fastq = is_valid_sample_fastq_file(sample_fastq, sample_internal_id="sample_id")
+
+    # THEN it should not be valid
+    assert not is_valid_fastq
+
+
+def test_validate_sample_fastq_with_invalid_file_extension():
+    # GIVEN a sample fastq file without a valid file extension
+    sample_fastq = Path("Sample_123/123_L0002.fastq")
+
+    # WHEN validating the sample fastq file
+    is_valid_fastq = is_valid_sample_fastq_file(sample_fastq, sample_internal_id="123")
+
+    # THEN it should not be valid
+    assert not is_valid_fastq
+
+
+def test_is_file_path_compressed_fastq_with_valid_file():
+    # GIVEN a valid .fastq.gz file
+    file_path = Path("sample_L0002.fastq.gz")
+
+    # WHEN checking if the file path is a compressed fastq file
+    result = is_file_path_compressed_fastq(file_path)
+
+    # THEN the result should be True
+    assert result is True
```

Comment:
```suggestion
    assert is_file_compressed
```
Similarly for the rest of module tests if you agree.